### ISSUE TITLE
Debug logs are absent despite DEBUG log level is enabled in release 2.32 builds

### ISSUE THREAD

**Issue by pavloburykh** [status-im/status-react#20944]
**Steps:** 

1. Install release 2.32 type of build
2. Create/recover a user
3. Change log level to DEBUG (Profile - Advanced - Log level - DEBUG)
4. Download the logs 
5. Check if there are debug logs in downloaded files
6. Check if geth log is there

**Actual result:** geth.log is absent at all. There are not debug logs in Status log file. 

Example of logs 

[logs (54).zip](https://github.com/user-attachments/files/18095267/logs.54.zip)

**Expected result:** geth.log should be in log folder. all log files should contain debug logs. 

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

**Comment by pavloburykh** [comment]
Hey @vkjr! Wondering if there are there any updates on this issue? No pressure from my side, just asking, maybe you need some additional help on this. We are currently in the middle of release testing and constantly facing backend issues. Because of current issue we unfortunately not able to share any valuable DEBUG logs to devs so they can investigate those bugs. So current issue is kinda blocker to us from this perspective.

**Comment by vkjr** [comment]
@pavloburykh, yes there are updates. This bug is related to `status-go` and I need an additional opinion.

@qfrank, in this [PR](https://github.com/status-im/status-go/pull/3643) you implemented `InitLogging` in a way that it should be called only once using this mechanism:
```
var initLoggingOnce sync.Once
```
As a result `InitLogging` works once when the app starts and uses the default log level. But when a specific user logs in, `InitLogging` called second time with user-specific log level, but that has no effect because of that call-once mechanism.
Are there any drawbacks in allowing `InitLogging` and nested `OverrideRootLogWithConfig` to be called an arbitrary amount of times? Because if I remove that `initLoggingOnce.Do`, logging works as expected and can be changed from mobile side. But I may not be aware of some problems with re-initing logger.
Wdyt, can we remove that `call-once` approach?

**Comment by qfrank** [comment]
you can remove, using sync.Once was because I applied the [suggestion](https://github.com/status-im/status-go/pull/3643#discussion_r1236800288) from @Samyoul and I thought we will only need invoke it once. 
Could you update the comment on `InitLogging` as well when you create the corresponding PR? thanks in advance ❤️ 
@vkjr

**Comment by pavloburykh** [comment]
fixed by https://github.com/status-im/status-mobile/pull/20974

**Comment by pavloburykh** [comment]
Reopening the issue. As was found out during release 2.32 testing, that debug logs are absent even if DEBUG log level is enabled. Moreover, geth log file is absent at all

**Comment by ilmotta** [comment]
@qfrank @pavloburykh @churik:

The regression was introduced by commit f45f96975e5a64b8f2378ce455989ea2cbff6690 in PR https://github.com/status-im/status-mobile/pull/21677. The parent commit 53d5ac4ea2af97f3814bf79e4acc34e046887626 works.

- The expected behavior before the regression was to create the `geth.log` file when the app starts, independently of the default log level, even if it was blank.
- In the PR which introduced the regression the `geth.log` file is only created if `status-im.config/log-level` is not blank, but the problem is that in release builds the log level is always blank. We don't want to default to `ERROR` because right now we can't guarantee private data won't leak.

This issue is a release blocker because not fixing it means we won't be able to get any output from `geth.log` in release 2.32.

**Comment by churik** [comment]
Should we get the fixes to `develop` @qfrank ? 
Can skip manual testing.
Thank you!

**Comment by qfrank** [comment]
Unfortunately, @igor-sirotin [doesn't agree](https://github.com/status-im/status-go/pull/6202#pullrequestreview-2502170310) we do it like this for `develop` @churik

**Comment by churik** [comment]
ah, got it, thank you, let's keep it open then
